deploy/docker/utils/bin/check_replica_set.js

```diff
@@ -0,0 +1,46 @@
+const { MongoClient } = require("mongodb");
+
+async function exec() {
+  const client = new MongoClient(process.env.APPSMITH_MONGODB_URI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+  });
+  checkReplicaSet(client).then((res) => {
+    // support replica set
+    if (res === 0) {
+      client.close();
+      process.exit(0);
+    }
+
+    // not support replica set
+    if (res === 1) {
+      client.close();
+      process.exit(1);
+    }
+  });
+}
+
+async function checkReplicaSet(client) {
+  await client.connect();
+  return new Promise((resolve) => {
```

Comment:
I think this will result in an unintentional nested-promise. We likely need an `await` here, like:

```suggestion
  return await new Promise((resolve) => {
```

Reply:
updated

Reply:
I don't see the `await` keyword in the return statement still, do you need to push your local commits perhaps? 🤔

Reply:
Sorry, my bad. I have updated from the local